### PR TITLE
Add documentation for onboarding a provider with metadata service

### PR DESCRIPTION
**Pull request by abhat**
This commit enhances the onboarding guide with necessary information
to secure a infrastructure provider's metadata service.

Signed-off-by: Aniket Bhat <anbhat@redhat.com>

### CODE REVIEW

**Review by elmiko — COMMENTED** [review]
thanks for the contribution @abhat !

i wonder if we should put this doc in the networking section?

@candita @Miciah any thoughts?

**Comment by candita** [comment]
@abhat @elmiko this is a little different than the existing network section, and certainly should have its own section in the overview.md.  My thoughts - the section in the overview should be named more like networking-ovn-sdn, and the metadata service be a topic within the networking-ovn-sdn  section.

File: docs/overview.md [hunk 1]
```diff
@@ -65,6 +65,14 @@ within the list reference documents with greater detail for each step.
     [Cluster Ingress Operator](https://github.com/openshift/cluster-ingress-operator)
     for the new infrastructure provider.
 
+1. [Securing Access to Metadata Service](../networking-metadata-service)
+
+    Evaluate provider-specific support for metadata service.
+	If applicable, update the
+    [OpenShift SDN](https://github.com/openshift/sdn) and 
+	[OVN Kubernetes](https://github.com/openshift/ovn-kubernetes) CNI Plugins
+    for the new infrastructure provider.
```
Comment (candita) [inline docs/overview.md:74]:
```suggestion
1. [Software Defined Network and OVN](../networking-ovn-sdn)

    Evaluate provider-specific support for software defined networking.
	If applicable, update the
    [OpenShift SDN](https://github.com/openshift/sdn) and 
	[OVN Kubernetes](https://github.com/openshift/ovn-kubernetes) CNI Plugins
    for the new infrastructure provider.
```

File: docs/networking-metadata-service/index.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Securing Access to Metadata Service
```
Comment (candita) [inline docs/networking-metadata-service/index.md:1]:
```suggestion
# Software Defined Network and OVN

## Securing Access to Metadata Service
```

File: docs/networking-metadata-service/index.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+### Evaluate if your cloud provider runs a metadata service
+
+If your cloud provider runs a metadata service, you would typically want pods created in
```
Comment (candita) [inline docs/networking-metadata-service/index.md:5]: Maybe provide an example of a metadata service?  Even after reading this, I don't know what it is.

**Comment by elmiko** [comment]
that's a good idea @candita , perhaps we could name the section "Software Defined and Container Networking" ? @abhat wdyt?

my only question is, will there be other information we will need to share about ovn/sdn, if this is going to be purely about metadata services then i would prefer to name the section "Infrastructure Metadata Services" to make it explicit.

**Comment by elmiko** [comment]
sorry @abhat i needed to setup the main branch in preparation for the migration to the openshift repository, would you mind rebasing your change and re-opening this PR please.

this repository is about to move to https://github.com/openshift/infrastructure-provider-onboarding-guide , i'm happy to keep your PR here for the moment since we have a discussion going, and then i can cherry-pick it to the new home.

File: docs/networking-metadata-service/index.md (hunk 3 above)
Comment (abhat) [reply docs/networking-metadata-service/index.md:5]: Will do.

**Comment by abhat** [comment]
Ok, that's fine. let me know 

> that's a good idea @candita , perhaps we could name the section "Software Defined and Container Networking" ? @abhat wdyt?
> 
> my only question is, will there be other information we will need to share about ovn/sdn, if this is going to be purely about metadata services then i would prefer to name the section "Infrastructure Metadata Services" to make it explicit.

That's exactly right - there is nothing networking specific going on here. Networking just happens to be the owner of mechanisms to restrict access to the metadata service. I will call it Infrastructure Metadata Services. and have a section called Securing Access To Metadata Services

**Comment by elmiko** [comment]
thanks @abhat , and thank you for putting up with our housekeeping =)

**Pull request closed without merging**